### PR TITLE
Refactor terraform templates to use baremetal_host module

### PR DESCRIPTION
**Pull request by osherdp**
Currently we have a couple of terraform templates, each for a different installation flow:
* One for most scenarios, including (but not limited) to: ipv4, ipv6, static-networking, defining multiple disks for olm operators like lso, bootstrap-in-place mode for installing SNO just via openshift installer, and more
* None platform scenario. In this case it has a different networking structure where masters and workers are split between several L3 networks
* infra-env template for late-binding scenarios

This PR should remove some of the existing duplication by creating a module for baremetal hosts, and including it in the larger template.
It (currently) doesn't solve some of the duplication that we have regarding networks, DNS configuration etc., but that should be a good start either way.

### CODE REVIEW

**Comment by osherdp** [comment]
/cc @eliorerz @tsorya @omertuc @rollandf 
hi guys, please review this work
mainly help me make sure the structure is still right for none-platform (@rollandf) and late-binding (@omertuc)

**Comment by osherdp** [comment]
/test e2e-metal-assisted-none-ipv6
(known minikube's etcd problem)

**Comment by osherdp** [comment]
/test e2e-metal-assisted-none-ipv6
(just trying to understand if it's a flake)

File: discovery-infra/test_infra/tools/static_network.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def generate_macs(count: int) -> List[str]:
 
 
 def generate_day2_static_network_data_from_tf(tf_folder: str, num_day2_workers: int) -> List[Dict[str, List[dict]]]:
-    tfvars_json_file = os.path.join(tf_folder, consts.TFVARS_JSON_NAME)
+    tfvars_json_file = os.path.join(tf_folder, "baremetal", consts.TFVARS_JSON_NAME)
```
Comment (eliorerz) [inline discovery-infra/test_infra/tools/static_network.py:22]:
```suggestion
    tfvars_json_file = os.path.join(tf_folder, consts.Platforms.BARE_METAL, consts.TFVARS_JSON_NAME)
```

File: discovery-infra/test_infra/tools/static_network.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def generate_day2_static_network_data_from_tf(tf_folder: str, num_day2_workers:
 
 
 def generate_static_network_data_from_tf(tf_folder: str) -> List[Dict[str, List[dict]]]:
-    tfvars_json_file = os.path.join(tf_folder, consts.TFVARS_JSON_NAME)
+    tfvars_json_file = os.path.join(tf_folder, "baremetal", consts.TFVARS_JSON_NAME)
```
Comment (eliorerz) [inline discovery-infra/test_infra/tools/static_network.py:46]:
```suggestion
    tfvars_json_file = os.path.join(tf_folder, consts.Platforms.BARE_METAL, consts.TFVARS_JSON_NAME)
```

File: discovery-infra/test_infra/utils/kubeapi_utils.py [hunk 3]
```diff
@@ -64,7 +65,9 @@ def get_libvirt_nodes_from_tf_state(network_names: Union[List[str], Tuple[str]],
 
 def get_nodes_details(cluster_name, namespace, tf):
     tf_folder = utils.get_tf_folder(cluster_name, namespace)
-    tf_vars = utils.get_tfvars(tf_folder)
+    baremetal_template = os.path.join(tf_folder, "baremetal")
```
Comment (eliorerz) [inline discovery-infra/test_infra/utils/kubeapi_utils.py:68]: consts.Platforms.BARE_METAL

File: discovery-infra/test_infra/utils/utils.py [hunk 4]
```diff
@@ -504,15 +506,8 @@ def get_openshift_release_image(allow_default=True):
...
-def copy_template_tree(dst, none_platform_mode=False, is_infra_env=False):
-    copy_tree(
-        src=consts.TF_TEMPLATE_BARE_METAL_INFRA_ENV_FLOW
-        if is_infra_env
-        else consts.TF_TEMPLATE_NONE_PLATFORM_FLOW
-        if none_platform_mode
-        else consts.TF_TEMPLATE_BARE_METAL_FLOW,
-        dst=dst,
-    )
+def copy_template_tree(dst):
```
Comment (eliorerz) [inline discovery-infra/test_infra/utils/utils.py:509]: Add type annotations

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 5]
```diff
@@ -49,12 +49,15 @@ def _create_tf_folder(self, name: str, platform: str):
...
-        utils.copy_template_tree(
-            tf_folder,
-            none_platform_mode=(platform == consts.Platforms.NONE),
-            is_infra_env=isinstance(self._entity_config, BaseInfraEnvConfig),
-        )
-        return tf_folder
+        utils.copy_template_tree(tf_folder)
+
+        if platform == consts.Platforms.NONE:
+            return os.path.join(tf_folder, "none")
```
Comment (eliorerz) [inline discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:55]:
```suggestion
            return os.path.join(tf_folder, consts.Platforms.NONE)
```

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 6]
```diff
@@ -49,12 +49,15 @@ def _create_tf_folder(self, name: str, platform: str):
...
+
+        if isinstance(self._entity_config, BaseInfraEnvConfig):
+            return os.path.join(tf_folder, "baremetal_infra_env")
+
+        return os.path.join(tf_folder, "baremetal")
```
Comment (eliorerz) [inline discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:60]:
```suggestion
        return os.path.join(tf_folder, consts.Platforms.BARE_METAL)
```

**Review by eliorerz — CHANGES_REQUESTED** [review]

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**